### PR TITLE
Add missing `last-column-id` to spec

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -1541,6 +1541,8 @@ components:
           properties:
             snapshot:
               $ref: '#/components/schemas/Snapshot'
+            last-column-id:
```
Comment (nastra) [inline open-api/rest-catalog-open-api.yaml:1544]: I think this should be added to to `AddSchemaUpdate` instead of `AddSnapshotUpdate`

**Comment by rdblue** [comment]
The reason why this wasn't in the original spec is because we can technically update it by getting the highest assigned ID from the new schema. Because it didn't strictly need to be part of the spec here, we omitted it. Still, we send this in the actual implementation because it simplifies the server side.

I'm ambivalent about whether we actually should include this in the spec. Keeping the spec simple is usually the best option. But I can see the argument for doing this. Maybe we should update the test service to no longer require the `last-column-id`.

FYI @bryanck.

**Comment by snazy** [comment]
(Side note: I no longer use the the rest-spec in my experiments - for various reasons - and use the JSON serialization from `iceberg-core`.)

But IMO the spec should exactly reflect the behavior (or better. vice versa). Otherwise, what's the point of having a _specification_, if the actual behavior is different. Users that do not use the JSON code in `iceberg-core` or the provided Python client will rely on the OpenAPI spec, potentially to generate `curl` requests or even generate their own client code  from that spec.

**Comment by Fokko** [comment]
@snazy Thanks for this PR, and I couldn't agree more that we need to keep the spec as close to the implementation as possible. I'm going to close this one now since https://github.com/apache/iceberg/pull/7445 has been merged.

**Pull request closed without merging**